### PR TITLE
Eslint all files with prettier

### PR DESCRIPTION
**Pull request by dgreif**
ESLint was previously set up to do the default file glob, which only finds `.js` files.  With all the awesome TypeScript conversion @colebemis has been doing, lint has been covering less and less files.  This PR does the following:
 * adds coverage back to _all_ files (`.js`, `.ts`, `.tsx`)
 * adds the prettier plugin into the eslint config (it was a dependency, but wasn't being used)
 * fixes all the related lint issues
 * adds `yarn lint-fix` which can be used to run prettier if your IDE isn't already configured to do so
 * updates all the eslint/prettier related dependencies.

Note: I split the base eslint fixes and the prettier fixes into two separate commits.  The base fixes probably deserve the closer look of the two.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "dist:transpile:cjs": "rm -rf lib && cross-env BABEL_MODULE=commonjs babel src --out-dir lib --extensions '.js,.ts,.jsx,.tsx'",
     "dist:transpile:esm": "rm -rf lib-esm && babel src --out-dir lib-esm --extensions '.js,.ts,.jsx,.tsx'",
     "dist:types": "rm -rf types && tsc",
-    "lint": "eslint src docs/components",
+    "lint": "eslint '**/*.{js,ts,tsx}'",
+    "lint-fix": "eslint '**/*.{js,ts,tsx}' --fix",
```
Comment (colebemis) [inline package.json:16]: [nit] I'm not sure if we have a naming convention for npm scripts, but `lint:fix` seems to be more consistent with the other scripts than `lint-fix`.

```suggestion
    "lint:fix": "eslint '**/*.{js,ts,tsx}' --fix",
```

**Review by colebemis — COMMENTED** [review]
Thank you for doing this, @dgreif! ❤️ 

These changes look great. However, I tried running `yarn lint` locally on this branch and my computer fans started running on high and the script never finished. When I change the script to `eslint 'src/**/*.{js,ts,tsx}'` instead of `eslint '**/*.{js,ts,tsx}'` the script runs as expected. It seems like the `ignorePatterns` field might not be working as expect (at least on my machine).

File: package.json (hunk 1 above)
Comment (dgreif) [reply package.json:16]: Great call, I should have looked at the others to see what the normal pattern was so far

**Comment by dgreif** [comment]
Hmm, there must be another build output dir that I missed.  @colebemis can you think of another spot that `.js` files are generated?  I'd prefer not to limit it to `src/...` because there are lots of other one-off js files that wouldn't get picked up.

**Review by colebemis — APPROVED** [review]
Thanks for taking the initiative to fix this, @dgreif 🚀

**Pull request merged**